lib/GenomeFileUtil/core/FastaGFFToGenome.py

```diff
@@ -160,15 +164,41 @@ def _gen_genome_json(self, params, input_gff_file, input_fasta_file):
             genome_type = "metagenome"
         else:
             genome_type = params.get('genome_type', 'isolate')
-        assembly_ref = self.au.save_assembly_from_fasta(
-            {'file': {'path': input_fasta_file},
-             'workspace_name': params['workspace_name'],
-             'assembly_name': params['genome_name'] + ".assembly",
-             'type': genome_type,
-             })
-        assembly_data = self.dfu.get_objects(
-            {'object_refs': [assembly_ref],
-             'ignore_errors': 0})['data'][0]['data']
+        if params.get('existing_assembly_ref'):
+            assembly_ref = params['existing_assembly_ref']
+
+            if not re.match("\d+\/\d+\/\d+", assembly_ref):
```

Comment:
What about workspace refs that have the format "workspace_name/object_name". This format is valid for the workspace

Reply:
should I just let the DFU handle if the `assembly_ref` validity checking?